### PR TITLE
Allow adding and dropping fields to records wrapped in array

### PR DESCRIPTION
**Pull request by losipiuk**
## Description

Rework of https://github.com/trinodb/trino/pull/22115 following syntax guidelines from https://github.com/trinodb/trino/issues/16959

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
(x) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by losipiuk** [comment]
@martint I added `syntax-needs-review` label - but I am not sure if needed given https://github.com/trinodb/trino/issues/16959 exists

File: core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java [hunk 1]
```diff
@@ -166,6 +168,7 @@ public ListenableFuture<Void> execute(
             Type currentType = columnMetadata.getType();
             for (int i = 0; i < fieldPath.size() - 1; i++) {
                 String fieldName = fieldPath.get(i);
+
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java:171]: looks redundant

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java (hunk 1 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java:171]: In what sense?

Comment (findepi) [reply core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java:171]: redundant change

**Comment by losipiuk** [comment]
CI: https://github.com/trinodb/trino/issues/18697

File: core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java (hunk 1 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java:171]: Oh - new line - yeah :)

**Comment by losipiuk** [comment]
> @martint I added `syntax-needs-review` label - but I am not sure if needed given #16959 exists

ping @martint

**Comment by losipiuk** [comment]
> > @martint I added `syntax-needs-review` label - but I am not sure if needed given #16959 exists
> 
> ping @martint

ping @martint

**Comment by losipiuk** [comment]
> > > @martint I added `syntax-needs-review` label - but I am not sure if needed given #16959 exists
> > 
> > 
> > ping @martint
> 
> ping @martint

friendly ping @martint. This is simple - would be nice to close

**Review by martint — APPROVED** [review]

**Comment by losipiuk** [comment]
Thanks @martint  :)

**Pull request merged**